cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -383,18 +383,17 @@ public int hashCode() {
    * Id.Stream uniquely identifies a stream.
    */
   public static final class Stream {
-    private final String namespace;
+    private final Namespace namespace;
```

Comment:
no. The computation itself need to be fast, as you still computing it on every event (from the StreamHandler, you construct a new Id.Stream object and that will be used to lookup a Map, and that's when the hashCode is computed on the newly created Id.Stream object. The cache only saves the one in the Map already).